src/store/modules/layers.store.js

```diff
@@ -447,22 +448,30 @@ const actions = {
         }
         commit('updateLayer', updatedLayer)
     },
-    updateKmlLayer({ commit, getters, rootState }, payload) {
-        const { layerId, kmlData, kmlMetadata } = payload
+    updateKmlGpxLayer({ commit, getters, rootState }, payload) {
+        const { layerId, data, metadata } = payload
         const currentLayer = getters.getActiveLayerById(layerId)
         if (!currentLayer) {
             throw new Error(
-                `Failed to update KML layer data/metadata "${layerId}", ` +
+                `Failed to update GPX/KML layer data/metadata "${layerId}", ` +
                     `layer not found in active layers`
             )
         }
         const updatedLayer = currentLayer.clone()
 
-        if (kmlData) {
-            updatedLayer.name = parseKmlName(kmlData) || 'KML'
-            updatedLayer.kmlData = kmlData
+        if (data) {
+            let extent
+            if (updatedLayer.type === LayerTypes.KML) {
+                updatedLayer.name = parseKmlName(data) || 'KML'
+                updatedLayer.kmlData = data
+                extent = getKmlExtent(data)
+            } else if (updatedLayer.type === LayerTypes.GPX) {
```

Comment:
I think there is no way to enter this function with anything other than KML or GPX layers, we could get rid of one check by turning this into a simple else

That being said, this is a bit more resilient (in case a non GPX / KML layer enter this function) and easier to read. I would add a 'else' statement to have a  default comportment when entering with an other type of Layer, to prevent side effects to go untracked for a while.

Reply:
I thought about it too, but I think it's a better solution to keep the two `if`s so that a non GPX/KML layer would fail quickly and without going into some kind of default parsing